src/components/EditorMarkdownIt.vue

```diff
@@ -51,7 +55,49 @@ export default {
 	methods: {
 		onUpdate() {
 			this.html = this.md.render(this.value)
+			setTimeout(() => this.prepareOnChangeListener(), 100)
+		},
+
+		prepareOnChangeListener() {
+			if (this.readonly) {
+				return
+			}
+			const items = document.getElementsByClassName('task-list-item')
+			for (let i = 0; i < items.length; ++i) {
+				items[i].removeEventListener('click', this.setListener)
+				items[i].addEventListener('click', this.setListener)
+			}
+		},
+
+		setListener(item) {
+			let idOfCheckbox = 0
+			const markdownLines = this.value.split('\n')
+			markdownLines.forEach((line, i) => {
+				// Regex Source: https://github.com/linsir/markdown-it-task-checkbox/blob/master/index.js#L121
+				// plus the '- '-string.
+				if (/^(- )\[[xX \u00A0]\][ \u00A0]/.test(line.trim())) {
```

Comment:
```suggestion
				if (/^-\s+\[[xX \u00A0]\][ \u00A0]/.test(line.trim())) {
```
This did not work correctly if there are multiple spaces after the dash, e.g. for `-  [x] item`